p2p/p2p/src/lib.rs

```diff
@@ -54,7 +55,10 @@ where
         cuprate_address_book::init_address_book(config.address_book_config.clone()).await?;
     let address_book = Buffer::new(
         address_book,
-        config.max_inbound_connections + config.outbound_connections,
+        config
+            .max_inbound_connections
+            .checked_add(config.outbound_connections)
+            .unwrap(),
```

Comment:
This is small but I think overflows in these types of cases usually saturate in most programs instead of crashing for UX reasons (better to saturate on already unrealistic and irrelevantly large number over crashing on said irrelevant number).

User input causing overflow is unrealistic yet possible if someone does this in the CLI/config:
```
max_inbound_connections = 99999999999999999999
```
which I could imagine at least a few people doing. I think `cuprated` continuing with `MAX` over crashing would be expected, or... `monerod` catches large input during CLI/config in stuff like `--out-peers`, `--limit-rate`, etc.

```
$ ./monerod --out-peers 18446744073709551615
Failed to parse arguments: the argument ('18446744073709551615') for option '--out-peers' is invalid
```
If we're either saturating or early returning during CLI, these `unwrap`s can be removed.

Reply:
Saturating these values will cause memory exhaustion as the channel allocates  I'll keep this as a TODO to add limits when parsing the config.